### ISSUE TITLE
[1.4.x] fltk-config --use-images: not including libfltk_jpeg.a?

### ISSUE THREAD

**Issue by erco77** [fltk/fltk#689]
**Describe the bug**
I don't know if this is a bug or just it's late and I need to go to bed, but I have a project that calls fltk-config to build the tool and uses the `--use-images` to build, but on one of my machines it's giving these errors:
```
$ /usr/local/src/fltk-1.4.x.git/build/fltk-config --compile altimeter.cxx --use-images
/usr/bin/c++ -I/usr/local/src/fltk-1.4.x.git/build -I/usr/local/src/fltk-1.4.x.git -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -D_THREAD_SAFE -D_REENTRANT -o 'altimeter' 'altimeter.cxx' /usr/local/src/fltk-1.4.x.git/build/lib/libfltk_images.a -lpng -lz /usr/local/src/fltk-1.4.x.git/build/lib/libfltk.a -lm -lpthread -lXinerama -lXfixes -lXcursor -lpangocairo-1.0 -lpango-1.0 -lgobject-2.0 -lglib-2.0 -lharfbuzz -lcairo -lcairo -lgtk-3 -lgdk-3 -lgio-2.0 -lX11 -lXft -lXrender -lwayland-cursor -lwayland-client -lxkbcommon -ldbus-1 -lfontconfig -ldl
/usr/bin/ld: /usr/local/src/fltk-1.4.x.git/build/lib/libfltk_images.a(Fl_JPEG_Image.cxx.o): in function `jpeg_unprotected_mem_src(jpeg_decompress_struct*, unsigned char const*)':
Fl_JPEG_Image.cxx:(.text+0x2aa): undefined reference to `fltk_jpeg_resync_to_restart'
/usr/bin/ld: /usr/local/src/fltk-1.4.x.git/build/lib/libfltk_images.a(Fl_JPEG_Image.cxx.o): in function `Fl_JPEG_Image::load_jpg_(char const*, char const*, unsigned char const*, int)':
Fl_JPEG_Image.cxx:(.text+0x40e): undefined reference to `fltk_jpeg_std_error'
/usr/bin/ld: Fl_JPEG_Image.cxx:(.text+0x517): undefined reference to `fltk_jpeg_finish_decompress'
/usr/bin/ld: Fl_JPEG_Image.cxx:(.text+0x543): undefined reference to `fltk_jpeg_destroy_decompress'
/usr/bin/ld: Fl_JPEG_Image.cxx:(.text+0x64e): undefined reference to `fltk_jpeg_CreateDecompress'
[..etc..]
```

Seems there's a `lib/libfltk_jpeg.a` being generated by the build that `fltk-config --use-images` is not including in the link line.

When I copy/paste the c++ link line from the above and add the full path to the `libfltk_jpeg.a` file (after the `libfltk_images.a`) it builds OK.

Seems like fltk-config should be including this file automatically?

I don't think any of our default builds use `fltk-config` to build e.g. any of the test programs, so this perhaps has gone undetected.

**To Reproduce**
Build FLTK on a linux machine (might need a machine that doesn't have a local JPEG lib?) but don't install.
Then in a directory outside of the FLTK directory, try to use the `build/fltk-config` to build any simple fltk app that needs --use-images.

**Expected behavior**
No jpeg errors during link

**FLTK Version**
Current 1.4.x

**FLTK Configure / Build Options**
In my case just:
```
cmake ..
```
which at the end prints the summary as follows (**Note that last bit about JPEG being "Builtin" by default**)

```
-- Configuration Summary for FLTK 1.4.0 generated by CMake 3.18.4 --

-- The following OPTIONAL packages have been found:

 * Doxygen
 * LATEX
 * ZLIB
 * PNG

-- The following OPTIONAL packages have not been found:

 * JPEG

-- Static libraries will be built in /usr/local/src/fltk-1.4.x.git/build/lib
-- Shared libraries will not be built (set OPTION_BUILD_SHARED_LIBS=ON to build)
-- FLUID will be built in /usr/local/src/fltk-1.4.x.git/build/bin/fluid
-- fltk-options will be built in /usr/local/src/fltk-1.4.x.git/build/bin/fltk-options
-- Test programs    will be built in /usr/local/src/fltk-1.4.x.git/build/bin/test
-- Example programs will not be built (set FLTK_BUILD_EXAMPLES=ON to build)
-- Image Libraries: JPEG = Builtin
--                  PNG  = System
--                  ZLIB = System
-- Use Wayland:             Yes (when available at run-time)
-- All drawing uses Cairo:  Yes
-- Use Pango:               Yes
-- Fl_Cairo_Window support: No

-- End of Configuration Summary --

-- Configuring done
-- Generating done
-- Build files have been written to: /usr/local/src/fltk-1.4.x.git/build
```

**Linux/Unix Runtime, if applicable:**
I think this machine has both Wayland and X11.

**Comment by Albrecht-S** [comment]
Bug confirmed. It's definitely a `CMake` issue which doesn't generate instructions to link the "local" image libs when one or more builtin FLTK image libs are created, for instance with `OPTION_USE_SYSTEM_LIBJPEG:BOOL=OFF` or because the lib is not found on the system. The latter is very likely on a default Linux system where zlib and png development packages are installed but jpeg dev packages are not installed.

FTR: It's not Linux specific, I found the same issue when cross-compiling on Linux for Windows. I also tested the "classic" `autoconf/configure/make` build and found that it generates a correct `fltk-config` script that adds the local libraries to the build.

FWIW: I could fix the generated script by copying the relevant lines from the script generated by configure:
```diff
$ diff -u fltk-config build/debug/fltk-config | grep jpeg
-IMAGELIBS="-lfltk_jpeg -lpng -lz "
-STATICIMAGELIBS="$libdir/libfltk_jpeg.a -lpng -lz "
+IMAGELIBS="-ljpeg -lpng -lz"
+STATICIMAGELIBS="-ljpeg -lpng -lz"
```

This is of course only a workaround. I'll take care of this issue, together with other `CMake` / `fltk-config` related issues.
See also:
- https://github.com/fltk/fltk/issues/129
- https://github.com/fltk/fltk/issues/655
